platform-sdk/swirlds-common/src/main/java/com/swirlds/common/threading/CountUpLatch.java

```diff
@@ -0,0 +1,195 @@
+    private final Condition condition = lock.newCondition();
+
+    /**
+     * Create a new CountUpLatch with an initial count of 0.
+     */
+    public CountUpLatch() {
+        this(0);
+    }
+
+    /**
+     * Create a new CountUpLatch with the given initial count.
+     *
+     * @param initialCount the initial count
+     */
+    public CountUpLatch(final long initialCount) {
+        this.currentCount = new AtomicLong(initialCount);
+    }
+
+    /**
+     * Get the current count.
+     *
+     * @return the current count
+     */
+    public long getCount() {
+        return currentCount.get();
+    }
+
+    /**
+     * Increment the count by 1.
+     * <p>
+     * Methods that update the count are not mutually thread safe. Concurrent threads should never attempt to update the
+     * count. It is, however, safe for many threads to concurrently call methods that query the count.
+     */
+    public void increment() {
+        add(1);
+    }
+
+    /**
+     * Used to set the count in a way that will not allow the count to decrease.
+     */
+    private static final LongBinaryOperator SET_COUNT =
+            (final long previous, final long proposed) -> Math.max(proposed, previous);
+
+    /**
+     * Set the count to a higher value.
+     * <p>
+     * Methods that update the count are not mutually thread safe. Concurrent threads should never attempt to update the
+     * count. It is, however, safe for many threads to concurrently call methods that query the count.
+     *
+     * @param count the new count, must be greater than the current value
+     * @throws IllegalArgumentException if the new count is less than the current count (the state of this object is
+     *                                  messed up if this happens -- this is an unrecoverable error)
+     */
+    public void set(final long count) {
+        final long result = currentCount.accumulateAndGet(count, SET_COUNT);
+        if (result != count) {
+            throw new IllegalArgumentException(
+                    "Can't set the count to a lower value. Previous = " + result + ", provided = " + count);
+        }
+        lock.lock();
+        try {
+            condition.signalAll();
+        } finally {
+            lock.unlock();
+        }
+    }
+
+    /**
+     * Increment the count by the given delta.
+     * <p>
+     * Methods that update the count are not mutually thread safe. Concurrent threads should never attempt to update the
+     * count. It is, however, safe for many threads to concurrently call methods that query the count.
+     *
+     * @param delta the amount to increment the count by
+     * @throws IllegalArgumentException if the delta is negative
+     */
+    public void add(final long delta) {
+        if (delta < 0) {
+            throw new IllegalArgumentException("Cannot add a negative delta (" + delta + ")");
+        }
+        currentCount.addAndGet(delta);
+
+        lock.lock();
```

Comment:
we acquire a lock every time we increment the count? this is not very performant, I thought we wanted to avoid this

Reply:
I suggest we rather go with a busy wait than this

Reply:
We will call this method on the order of once per second. (More specifically, once per round plus once per file close operation.) We discussed this, but repeating it here for others to observe.

Reply:
Resolving this based on our dicussion